NodePackageTool.hs

```diff
@@ -149,12 +150,24 @@ realMain Args{..} = do
     -- | Link a binary file to @targetDir/name@.
     -- @relBinPath@ is relative from the package dir.
     linkBin :: FilePath -> (Text, FilePath) -> ErrorLogger ()
-    linkBin targetDir_ (name, relBinPath) = do
+    linkBin targetDir_ (name_, relBinPath) = do
       binPath <- canonPkg relBinPath
-      targetDir <- canon targetDir_
-      tryIOMsg
-        (\e -> "Symlink could not be created: " <> e)
-        (PosixFiles.createSymbolicLink binPath $ targetDir FP.</> toS name)
+      (name, targetDir) <- traverse canon $
+        symlinkTarget name_ targetDir_
+      tryIOMsg (\e -> "Symlink (or directory) could not be created: " <> e) $ do
+        Dir.createDirectoryIfMissing False targetDir
+        PosixFiles.createSymbolicLink binPath
+          $ targetDir FP.</> toS name
```

Comment:
Let’s split up the tryIOMsg into two separate error messages.

Reply:
Addressed in c9cc1c38aaf03d1efa83fb32928c0b8a4beb6dd0